docs/MOCK_MODE.md

```
@@ -0,0 +1,94 @@
+# Mock Mode Documentation
+
+MadSpark Multi-Agent System includes a comprehensive mock mode that allows you to use the system without API keys. This is perfect for development, testing, and trying out the system.
+
+## How Mock Mode Works
+
+### Automatic Detection
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)